src/vascMorphology.cpp

```diff
@@ -0,0 +1,147 @@
+#include <cassert>
+#include <iostream>
+#include <unistd.h>
+
+#include <fstream>
+#include <streambuf>
+
+#include <morphio/vascMorphology.h>
+#include <morphio/vascSection.h>
+
+#include "plugin/vasc_morphologyHDF5.h"
+#include "plugin/morphologySWC.h"
+
+namespace morphio {
+namespace vasculature
+{
+
+void buildConnectivity(std::shared_ptr<property::Properties> properties);
+
+VasculatureMorphology::VasculatureMorphology(const morphio::URI &source, unsigned int options)
+{
+    const size_t pos = source.find_last_of(".");
+    if (pos == std::string::npos)
+        LBTHROW(UnknownFileType("File has no extension"));
+
+    if (access(source.c_str(), F_OK) == -1)
+        LBTHROW(RawDataError("File: " + source + " does not exist."));
+
+    std::string extension;
+
+    for (auto& c : source.substr(pos))
+        extension += std::tolower(c);
+
+    auto loader = [&source, &options, &extension]() {
+        if (extension == ".h5")
+            return plugin::h5::VasculatureMorphologyHDF5().load(source);
+        //if (extension == ".swc")
+        //    return plugin::swc::load(source, options);
+        LBTHROW(UnknownFileType(
+                "Unhandled file type"));
+    };
+
+    _properties = std::make_shared<property::Properties>(loader());
+
+    buildConnectivity(_properties);
+}
+
+VasculatureMorphology::VasculatureMorphology(VasculatureMorphology&&) = default;
+VasculatureMorphology& VasculatureMorphology::operator=(VasculatureMorphology&&) = default;
+
+VasculatureMorphology::~VasculatureMorphology() {}
+
+bool VasculatureMorphology::operator==(const VasculatureMorphology& other) const
+{
+    if (this->_properties == other._properties)
```

Comment:
It already returns a boolean, so why not `return this->_properties == other._properties`